FlowCrypt/Controllers/Compose/ComposeViewController.swift

```diff
@@ -724,46 +764,94 @@ extension ComposeViewController {
         }
     }
 
-    private func recipientsNode() -> RecipientEmailsCellNode {
+    private func recipientsNode(type: RecipientType) -> ASCellNode {
+        let recipients = contextToSend.recipients(type: type)
+
+        let shouldShowToggleButton = type == .to
+            && recipients.isNotEmpty
+            && !contextToSend.hasCcOrBccRecipients
+
         return RecipientEmailsCellNode(
             recipients: recipients.map(RecipientEmailsCellNode.Input.init),
-            height: calculatedRecipientsPartHeight ?? Constants.minRecipientsPartHeight
-        )
+            type: type.rawValue,
+            height: recipientsNodeHeight(type: type) ?? Constants.minRecipientsPartHeight,
+            isToggleButtonRotated: shouldShowAllRecipientTypes,
+            toggleButtonAction: shouldShowToggleButton ? { [weak self] in
+                guard type == .to else { return }
+                self?.toggleRecipientsList()
+            } : nil)
             .onLayoutHeightChanged { [weak self] layoutHeight in
-                guard self?.calculatedRecipientsPartHeight != layoutHeight, layoutHeight > 0 else {
-                    return
-                }
-                self?.calculatedRecipientsPartHeight = layoutHeight
+                self?.updateRecipientsNode(
+                    layoutHeight: layoutHeight,
+                    type: type
+                )
             }
             .onItemSelect { [weak self] (action: RecipientEmailsCellNode.RecipientEmailTapAction) in
                 switch action {
                 case let .imageTap(indexPath):
-                    self?.handleRecipientAction(with: indexPath)
+                    self?.handleRecipientAction(with: indexPath, type: type)
                 case let .select(indexPath):
-                    self?.handleRecipientSelection(with: indexPath)
+                    self?.handleRecipientSelection(with: indexPath, type: type)
                 }
             }
     }
 
-    private func recipientInput() -> TextFieldCellNode {
-        TextFieldCellNode(
+    private func recipientsNodeHeight(type: RecipientType) -> CGFloat? {
+        switch type {
+        case .to:
+            return calculatedRecipientsToPartHeight
+        case .cc:
+            return calculatedRecipientsCcPartHeight
+        case .bcc:
+            return calculatedRecipientsBccPartHeight
+        }
+    }
+
+    private func updateRecipientsNode(layoutHeight: CGFloat, type: RecipientType) {
+        let currentHeight = self.recipientsNodeHeight(type: type)
+
+        guard currentHeight != layoutHeight, layoutHeight > 0 else {
+            return
+        }
+
+        switch type {
+        case .to:
+            self.calculatedRecipientsToPartHeight = layoutHeight
+        case .cc:
+            self.calculatedRecipientsCcPartHeight = layoutHeight
+        case .bcc:
+            self.calculatedRecipientsBccPartHeight = layoutHeight
+        }
+    }
+
```

Comment:
There's quite a bit of rendering or layouting code now in this VC, which makes business logic less obvious. Would it make sense to move it into some sort of decorator or rendering file (specific for this VC) instead of being here? Could be another PR, more of a maintenance task.